### PR TITLE
Feat/finalize guideline actions

### PR DESCRIPTION
**Pull request by GoranRibic**

### CODE REVIEW

File: packages/app-bridge/src/react/useGuidelineActions.ts [hunk 1]
```diff
@@ -173,16 +175,31 @@ export const useGuidelineActions = (appBridge: AppBridgeTheme) => {
     );
 
     const createPage = useCallback(
-        async (documentPage: CreateDocumentPage) => {
+        async (documentPage: DocumentPageRequestCreate) => {
```
Comment (SamuelAlev) [inline packages/app-bridge/src/react/useGuidelineActions.ts:178]:
```suggestion
        async (documentPage: CreateDocumentPage) => {
```
Why changing the name? It is less good than the previous one

File: packages/app-bridge/src/react/useGuidelineActions.ts [hunk 2]
```diff
@@ -146,7 +148,7 @@ export const useGuidelineActions = (appBridge: AppBridgeTheme) => {
     );
 
     const createDocumentGroup = useCallback(
-        async (documentGroup: CreateDocumentGroup) => {
+        async (documentGroup: Omit<DocumentGroupRequest, 'id'>) => {
```
Comment (SamuelAlev) [inline packages/app-bridge/src/react/useGuidelineActions.ts:151]: Please create a type instead, same for all others (`Omit<`, `RequireAtLeastOne<`)

File: packages/app-bridge/src/global.d.ts [hunk 3]
```diff
@@ -2,4 +2,14 @@
 
 type Nullable<T> = T | null;
 
+/**
+Type that behaves just like RequiredExactlyOne from type-fest with only difference that the remaining keys are set to optional.
+
+@category Object
+*/
+type RequireOnlyOne<ObjectType, KeysType extends keyof ObjectType = keyof ObjectType> = {
```
Comment (SamuelAlev) [inline packages/app-bridge/src/global.d.ts:10]: Isn't it `RequireAtLeastOne`?

File: packages/app-bridge/src/AppBridgeTheme.ts [hunk 4]
```diff
@@ -176,15 +212,39 @@ export class AppBridgeTheme {
         }
     }
 
-    public async createCoverPage(coverPage: CoverPageCreate) {
+    public async createDocumentCategory(category: Omit<DocumentCategoryRequest, 'id'>) {
```
Comment (SamuelAlev) [inline packages/app-bridge/src/AppBridgeTheme.ts:215]: Please create a type instead as we had before, same for all others

File: packages/app-bridge/src/AppBridgeTheme.ts [hunk 5]
```diff
@@ -176,15 +212,39 @@ export class AppBridgeTheme {
...
+
+    public async updateDocumentCategory(category: RequireAtLeastOne<DocumentCategoryRequest, 'title' | 'documentId'>) {
+        try {
+            return updateDocumentCategory(category as DocumentCategory);
+        } catch (error) {
+            throw console.error('Error: ', error);
+        }
+    }
+
+    public async deleteDocumentCategory(id: number) {
```
Comment (SamuelAlev) [inline packages/app-bridge/src/AppBridgeTheme.ts:231]:
```suggestion
    public async deleteDocumentCategoryById(id: number) {
```
And could you do it for all other `delete[...]`

For some models, we can delete by slug, by foreign id, ...

File: packages/app-bridge/src/AppBridgeTheme.ts [hunk 6]
```diff
@@ -19,148 +22,285 @@ import {
...
     }
 
-    public async createDocumentGroup(documentGroup: CreateDocumentGroup) {
-        return createDocumentGroup(documentGroup as DocumentGroup);
+    public async createDocumentPage(documentPage: DocumentPageRequestCreate) {
+        try {
+            return createDocumentPage({
+                ...documentPage,
+                ...(documentPage.linkUrl && { linkType: LinkType.External }),
+            } as DocumentPage);
```
Comment (SamuelAlev) [inline packages/app-bridge/src/AppBridgeTheme.ts:175]: If you need to cast you have wrong typing

File: packages/app-bridge/src/react/useGuidelineActions.ts (hunk 1 above)
Comment (SamuelAlev) [reply packages/app-bridge/src/react/useGuidelineActions.ts:178]: It makes it harder to read so I am up for dedicated types

File: packages/app-bridge/src/global.d.ts (hunk 3 above)
Comment (GoranRibic) [reply packages/app-bridge/src/global.d.ts:10]: Nope, as  RequireOnlyOne mutates other fields to be optional while RequireAtLeastOne leaves other keys as they are.

File: packages/app-bridge/src/types/DocumentPage.ts [hunk 7]
```diff
@@ -1,14 +1,29 @@
...
 export enum DocumentPageVisibility {
     Everyone = 'EVERYONE',
     Editor = 'EDITOR',
 }
 
+/**
+ * virtual fields of Document Page that wont be returned by API
+ */
+type DocumentPageApiVirtualFields = {
+    category_id?: Nullable<number>;
```
Comment (SamuelAlev) [inline packages/app-bridge/src/types/DocumentPage.ts:14]:
```suggestion
    document_category_id?: Nullable<number>;
```

File: packages/app-bridge/src/react/useDocumentPages.ts [hunk 8]
```diff
@@ -57,7 +55,68 @@ export const useDocumentPages = (appBridge: AppBridgeTheme, documentId: number)
...
+    const filteredPages = pages.filter((page) => page.id !== pageToDelete.id);
+
+    const updatedPages = filteredPages.map((page) => {
+        if ('documentPages' in page) {
+            page.documentPages = page.documentPages.filter((page) => page.id !== pageToDelete.id);
+        }
+        return page;
+    });
+
+    return updatedPages;
```
Comment (SamuelAlev) [inline packages/app-bridge/src/react/useDocumentPages.ts:121]:
```suggestion
    return filteredPages.map((page) => {
        if ('documentPages' in page) {
            page.documentPages = page.documentPages.filter((page) => page.id !== pageToDelete.id);
        }
        
        return page;
    });
```

File: packages/app-bridge/src/react/useDocuments.ts [hunk 9]
```diff
@@ -62,3 +60,51 @@ export const useDocuments = (appBridge: AppBridgeTheme) => {
...
+    const filteredDocuments = documents.filter((document) => document.id !== documentToDelete.id);
+
+    const updatedDocuments = filteredDocuments.map((document) => {
+        if ('documents' in document && document.documents !== null) {
+            document.documents = document.documents.filter((document) => document.id !== documentToDelete.id);
+        }
+        return document;
+    });
+
+    return updatedDocuments;
```
Comment (SamuelAlev) [inline packages/app-bridge/src/react/useDocuments.ts:109]:
```suggestion
    return filteredDocuments.map((document) => {
        if ('documents' in document && document.documents !== null) {
            document.documents = document.documents.filter((document) => document.id !== documentToDelete.id);
        }
        
        return document;
    });
```

**Pull request merged**